### DISCUSSION TITLE
Is there any advantage to using charge limit on Asus laptop?

### DISCUSSION THREAD

**Discussion by Kikimahe** [seerge/g-helper#1574] [Show and tell]
Hi

Don't be affraid Seerge, this post is not a problem with Ghelper which works very well. It's wonderful software. \o/ :-)

Just wanted to share an observation made with the 60% battery charge limit on my tuf 2022 12700H/32GB RAM/RTX 3060.

I tested a game (Humanitz) in CPU Turbo and GPU and I looked at the HWinfo64 sensors.

With the battery charge limit at 60%:
```
- Base clocks: 2.5 GHz on average
- CPU power: 28W on average
- FPS: 110fps
```

With the battery 100% charged:
```
Base clocks: 3.5 GHz on average
CPU power: +45W on average
FPS: 120 fps
```

I also did a benchmark with 3DMARK

With the battery charge limit at 60%:
 ![Battery 60%](https://github.com/seerge/g-helper/assets/142881210/5c20f489-7bea-4325-a41e-f44fb4f3d6a5)

With the battery 100% charged:
![Battery 100%](https://github.com/seerge/g-helper/assets/142881210/91b91f0b-682f-4ee7-a21d-c6d1bce0187b)

The result is confusing and now I know I have to make a choice between battery health or performance. ;-)
Thanks for reading me.

**Comment by AudreyAP** [comment]
The laptop will adjust performance based on the current charge state of the battery, as the battery is less able to provide high amounts of power when it is not fully charged. 

IIRC there should only be a performance difference when using battery. AC should be unaffected.

**Reply by Kikimahe** [reply]
IIRC ? :)

I agree with you @AudreyAP 
Performance should only decrease when the laptop is running on battery power.

But there should be no difference when:
1) Battery charge limit is enabled and charged to 60%
2) AND The laptop is plugged in

But in this case above, there is a decrease, for me and other people.

It seems that ASUS has not integrated the charge limit into the Bios. 
Performance is reduced if the battery is <100%, even if the laptop is plugged in. ;'(

**Reply by AudreyAP** [reply]
IIRC = if I remember correctly haha

Many laptops will use the battery as a "shock absorber" for very short periods of very high load while on ac(usually gpu spikes as Nvidia is known for), maybe that's why? Or maybe they just forgot to disable the performance limiters while connected to ac lol, I wouldn't put it past them. 

You may also want to make sure your bios is updated? Maybe fixed in new ones? Also, chipset drivers and Nvidia drivers? If you use nvcleanstall, make sure you enable Nvidia platform controllers.

**Reply by Kikimahe** [reply]
Of course, bios and drivers are up to date. ;-)

On another forum, someone gave me a workaround :
On Ghelper, use a CPU limit of 40W and a GPU limit of 100W (1700-1800 MHz).
The objective is not to exceed 146W which would be the limit before the CPU decreases to -30W.

It works. 
Performance is not maximum in games, but it remains consistent with the same FPS. :)
